### PR TITLE
[single-new] Remove dependency over pthread

### PR DESCRIPTION
**Pull request by kparichay**
Remove the dependency over pthread
Use glib functionalities instead

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single-new.c [hunk 1]
```diff
@@ -440,14 +431,16 @@ ml_single_open (ml_single_h * single, const char *model,
...
   single_h->join = FALSE;
 
-  if (pthread_create (&single_h->thread, NULL, invoke_thread,
-          (void *) single_h) < 0) {
-    ml_loge ("Failed to create the invoke thread.");
+  single_h->thread = g_thread_try_new (NULL, invoke_thread, (gpointer) single_h,
+      &error);
+  if (single_h->thread == NULL) {
+    ml_loge ("Failed to create the invoke thread, error: %s.", error->message);
+    g_free (error);
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single-new.c]: plz use g_clear_error (&error);

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: updated.

File: api/capi/src/nnstreamer-capi-single-new.c [hunk 2]
```diff
@@ -478,10 +471,10 @@ ml_single_close (ml_single_h single)
   ML_SINGLE_GET_VALID_HANDLE_LOCKED (single_h, single, 1);
 
   single_h->join = TRUE;
-  pthread_cond_broadcast (&single_h->cond);
+  g_cond_broadcast (&single_h->cond);
 
   ML_SINGLE_HANDLE_UNLOCK (single_h);
-  pthread_join (single_h->thread, NULL);
+  g_thread_join (single_h->thread);
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single-new.c]: plz consider to add g_mutex_clear() and g_cond_clear() before freeing single-handle.

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: updated.

**Pull request merged**